dfc/httpcommon.go

```diff
@@ -667,8 +667,13 @@ func (h *httprunner) setconfig(name, value string) (errstr string) {
 //=================
 func (h *httprunner) invalmsghdlr(w http.ResponseWriter, r *http.Request, specific string, other ...interface{}) {
 	status := http.StatusBadRequest
+	shouldWriteError := true
 	if len(other) > 0 {
-		status = other[0].(int)
+		if v, ok := other[0].(int); ok {
```

Comment:
this probably not a good idea, the API changed from (w, r, errstr, status) to (w, r, errstr, bool).
have you checked all caller of invalmsghdlr()? are they all called with one or none parameters? if that's the case, which I think it is, based on the fact that it only checked other[0], then I would make this more explicit, as (w, r, errstr, status, writeErr), seems a lot change though.
or keep it this way and document it clearly.
the existing API only try to avoid passing status, explicit calls seems better, a wrapper like can take care of them
and the name invalmsghdlr is really bad to me.
but any way, you can keep it your way, from a bad API to slight worse API, fine.